Source/Laser/LaserParticleContainer.cpp

```diff
@@ -52,7 +52,21 @@ LaserParticleContainer::LaserParticleContainer (AmrCore* amr_core, int ispecies,
 
     pp.query("pusher_algo", m_pusher_algo);
     getWithParser(pp, "wavelength", m_wavelength);
-    getWithParser(pp, "e_max", m_e_max);
+    const bool e_max_is_specified = queryWithParser(pp, "e_max", m_e_max);
+    Real a0;
+    const bool a0_is_specified = queryWithParser(pp, "a0", a0);
+    if (a0_is_specified){
+        Real omega = 2*MathConst::pi*PhysConst::c/m_wavelength;
+        m_e_max = PhysConst::m_e * omega * PhysConst::c * a0 / PhysConst::q_e;
+        if (e_max_is_specified){
+            amrex::Print()<<"Warning: both e_max and a0 specified. a0 overrides.\n";
+        }
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        e_max_is_specified || a0_is_specified,
```

Comment:
We could make this an xor:
```suggestion
        e_max_is_specified ^ a0_is_specified,
```
We want one of them specified, but not both (because then we would need to document and handle precedence).

Reply:
Nice, this could indeed be used in a few places to clean things up.